sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala

```diff
@@ -278,4 +281,14 @@ class QueryExecutionErrorsSuite extends QueryTest
     assert(e.getMessage ===
       "Datetime operation overflow: add 1000000 YEAR to TIMESTAMP '2022-03-09 01:02:03'.")
   }
+
+  test("CANNOT_PARSE_DECIMAL: unparseable decimal") {
+    val e = intercept[SparkIllegalStateException] {
+      val decimalParser = ExprUtils.getDecimalParser(Locale.ROOT)
+      decimalParser("$92,807.99")
```

Comment:
Could you raise the exception from user space (not using internal methods), please. We assign error classes not for user visible errors, just in case.

Reply:
I do it, pls review again！Unusually deep exception